src/haz3lcore/dynamics/elaborator.re

```diff
@@ -199,6 +199,11 @@ let rec dhexp_of_uexp = (m: Statics.map, uexp: Term.UExp.t): option(DHExp.t) =>
         /* not recursive */
         let* dp = dhpat_of_upat(m, p);
         let* ddef = dhexp_of_uexp(m, def);
+        let ddef =
+          switch (ddef) {
+          | Fun(a, b, c, _) => DHExp.Fun(a, b, c, Term.UPat.get_fun_name(p))
+          | _ => ddef
```

Comment:
need to handle the case of a function in Parens here as well

Reply:
<img width="475" alt="image" src="https://user-images.githubusercontent.com/73775425/198751707-da444062-c3fb-42e7-86ba-6dd5949263a4.png">

I think it handled the case of a function in Parens now.